crates/simulacrum/src/store.rs

```diff
@@ -0,0 +1,373 @@
+                .entry(object_id)
+                .or_default()
+                .insert(version, object);
+        }
+    }
+}
+
+impl BackingPackageStore for InMemoryStore {
+    fn get_package_object(
+        &self,
+        package_id: &ObjectID,
+    ) -> sui_types::error::SuiResult<Option<Object>> {
+        Ok(self.get_object(package_id).cloned())
+    }
+}
+
+impl ChildObjectResolver for InMemoryStore {
+    fn read_child_object(
+        &self,
+        parent: &ObjectID,
+        child: &ObjectID,
+        child_version_upper_bound: SequenceNumber,
+    ) -> sui_types::error::SuiResult<Option<Object>> {
+        let child_object = match self.get_object(child).cloned() {
+            None => return Ok(None),
+            Some(obj) => obj,
+        };
+
+        let parent = *parent;
+        if child_object.owner != Owner::ObjectOwner(parent.into()) {
+            return Err(SuiError::InvalidChildObjectAccess {
+                object: *child,
+                given_parent: parent,
+                actual_owner: child_object.owner,
+            });
+        }
+
+        if child_object.version() > child_version_upper_bound {
+            return Err(SuiError::UnsupportedFeatureError {
+                error: "TODO InMemoryStorage::read_child_object does not yet support bounded reads"
+                    .to_owned(),
+            });
+        }
+
+        Ok(Some(child_object))
+    }
+}
+
+impl GetModule for InMemoryStore {
+    type Error = SuiError;
+    type Item = CompiledModule;
+
+    fn get_module_by_id(&self, id: &ModuleId) -> Result<Option<Self::Item>, Self::Error> {
+        Ok(self
+            .get_module(id)?
+            .map(|bytes| CompiledModule::deserialize_with_defaults(&bytes).unwrap()))
+    }
+}
+
+impl ModuleResolver for InMemoryStore {
+    type Error = SuiError;
+
+    fn get_module(&self, module_id: &ModuleId) -> Result<Option<Vec<u8>>, Self::Error> {
+        Ok(self
+            .get_package(&ObjectID::from(*module_id.address()))?
+            .and_then(|package| {
+                package
+                    .serialized_module_map()
+                    .get(module_id.name().as_str())
+                    .cloned()
+            }))
+    }
+}
+
+impl ObjectStore for InMemoryStore {
+    fn get_object(
+        &self,
+        object_id: &ObjectID,
+    ) -> Result<Option<Object>, sui_types::error::SuiError> {
+        Ok(self.get_object(object_id).cloned())
+    }
+
+    fn get_object_by_key(
+        &self,
+        object_id: &ObjectID,
+        version: sui_types::base_types::VersionNumber,
+    ) -> Result<Option<Object>, sui_types::error::SuiError> {
+        Ok(self.get_object_at_version(object_id, version).cloned())
+    }
+}
+
+impl ParentSync for InMemoryStore {
+    fn get_latest_parent_entry_ref_deprecated(
+        &self,
+        _object_id: ObjectID,
+    ) -> sui_types::error::SuiResult<Option<sui_types::base_types::ObjectRef>> {
+        panic!("Never called in newer protocol versions")
```

Comment:
Any plans to fix this?

Reply:
yes we can fix/address this if we plan to use this with older protocol versions.